kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+		}
+
+		retval = taxReporting1099Dao.getVendors(vendorOwnershipCode, vendorOwnershipCategoryCode, isAllow);
+
+		if (LOG.isInfoEnabled()) {
+			int cnt = 0;
+			if ((retval != null) && !retval.isEmpty()) {
+				cnt = retval.size();
+			}
+
+			LOG.info("found " + cnt + " vendors with ownership code=" + vendorOwnershipCode);
+		}
+
+		return retval;
+	}
+
+	private PdfReader getPdfReader(Integer year) {
+		PdfReader retval = null;
+		FileInputStream fispdf = null;
+		try {
+			fispdf = new FileInputStream(get1099PdfFile(year));
+			retval = new PdfReader(fispdf);
+		} 
+		catch (Exception ex) {
+			throw new RuntimeException(ex);
+		}
+
+		finally {
+			try {
+				fispdf.close();
+			} 
+			catch (Exception ex) {
+			}
+
+		}
+
+		return retval;
+	}
+
+	private File get1099PdfFile(Integer year) {
+		File retval = null;
+
+		String fname = pdfDirectory + "/" + TaxPropertyConstants.PDF_1099_FILENAME_PREFIX + "-" + year + ".pdf";
+
+		LOG.info("pdfDirectory: " + pdfDirectory);
+
+		retval = new File(fname);
+
+		if (!retval.exists() || !retval.isFile() || retval.length() <= 0) {
+			throw new RuntimeException("The IRS 1099 Misc PDF [" + retval.getAbsolutePath() + "] is not available in the file system.");
+		}
+
+		LOG.info("1099 PDF filename = " + retval.getAbsolutePath());
+
+		return retval;
+	}
+
+	@Override
+	public List<Payee> loadPayees(Integer year) {
+		Map<String, Object> criteria = new HashMap<String, Object>();
+		criteria.put(TaxPropertyConstants.TAX_YEAR, year);
+		criteria.put(TaxPropertyConstants.EXCLUDE_INDICATOR, Boolean.FALSE);
+		return loadPayee1099Information(year, (List<Payee>) businessObjectService.findMatching(Payee.class, criteria));
+	}
+
+	private Payee loadPayee(Integer year, Payee payee) {
+		Payee retval = null;
+
+		// lets get all the payees for this a
+		Map<String, Object> criteria = new HashMap<String, Object>();
+
+		criteria.put(TaxPropertyConstants.TAX_YEAR, year);
+		criteria.put(TaxPropertyConstants.EXCLUDE_INDICATOR, Boolean.FALSE);
+		criteria.put(VendorPropertyConstants.VENDOR_HEADER_GENERATED_ID, payee.getVendorHeaderGeneratedIdentifier());
+		List<Payee> payees = (List<Payee>) businessObjectService.findMatching(Payee.class, criteria);
+
+		List<Payee> payeeInfo = loadPayee1099Information(year, payees);
+
+		if (!payeeInfo.isEmpty()) {
+			if (LOG.isDebugEnabled()) {
+				for (Payee p : payeeInfo) {
+					LOG.debug("payee: " + p.getVendorNumber());
+				}
+			}
+			retval = payeeInfo.get(0);
+		}
+
+		if (LOG.isDebugEnabled()) {
+			if (retval != null) {
+				LOG.debug("retval payee: " + retval.getVendorNumber());
+			} 
+			else {
+				LOG.debug("retval payee: is null");
+			}
+		}
+
+		return retval;
+	}
+
+	private List<Payee> loadPayee1099Information(Integer year, List<Payee> payees) {
+		List<Payee> retval = new ArrayList<Payee>();
+
+		if ((payees == null) || payees.isEmpty()) {
+			LOG.error("No payee candidates found");
```

Comment:
This isn't a fatal error in the logic (0 payee candidates are a possibility), the original coder chose to log this at the error level, so that if "something bad" happens during batch processing, it can be seen in the logs as a possible issue (I assume).